libs/platform/user/ebpf_extension_user.c

```diff
@@ -217,12 +217,15 @@ ebpf_provider_load(
 
 Done:
     ebpf_lock_unlock(&_ebpf_provider_table_lock, state);
-    ebpf_free(local_extension_provider);
+    // Wrap in conditional to resolve C6101.
```

Comment:
https://docs.microsoft.com/en-us/cpp/code-quality/c6101?view=msvc-160  says C6101 means "A successful path through the function does not set the named _Out_ parameter." which seems unrelated to 222.  Is C6101 the correct number and if so do you understand what it was complaining about?

Reply:
I think what is happening is analysis sees the local_extension_provider as an alias for the out parameter and incorrectly sees the ebpf_free as making it invalid. Wrapping it makes analysis see the out as still valid.

Reply:
Here's the full error text:
E:\ebpf-for-windows\libs\platform\user\ebpf_extension_user.c(159): warning C6101: Returning uninitialized memory '*local_extension_provider'.  A successful path through the function does not set the named _Out_ parameter. 

Note:
1) local_extension_provider is not an out parameter.
2) When the free occurs on the success path, local_extension_provider == NULL.

Reply:
![image](https://user-images.githubusercontent.com/20480683/123825211-f40e6480-d8bb-11eb-9222-a9dd591db747.png)

Here is the detailed error message from the analysis output.